ci(circle): add react node cache key suffix

loksonarius opened the pull request:
## Description

<!---
Lead with the intended commit body in this description field. For breaking
changes, please include "BREAKING CHANGE:" at the beginning of your commit
body.  At a minimum, this section should include a bracketed reference to the
Jira ticket, e.g. "[DET-1234]". When squash-and-merging, copy this directly
into the description field.
-->



## Test Plan

<!---
Describe the situations in which you've tested your change, and/or a screenshot
as appropriate.  Reviewers may ask questions about this test plan to ensure
adequate manual coverage of changes.
-->



## Commentary (optional)

<!---
Use this section of your description to add context to the PR. Could be for
particularly tricky bits of code that could use extra scrutiny, historical
context useful for reviewers, etc.
You may intentionally leave this section blank and remove the title.
--->



## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
<!---
Retain the relevant line and replace 000 with ticket number.

DET-000
MLG-000
WEB-000
DESIGN-000
No Ticket
--->


<!---
## Title

Example title: "docs: tweak recommended "pip install" usage".

Specifically, this title should contain a type and a description
of the change being made:

User-facing change types:

- docs: docs-only change
- feat: new user-facing feature
- fix: bug fix
- perf: performance improvement

Internal change types:

- build: build system change (anything in a `Makefile`, mostly)
- chore: any internal change not covered by another type
- ci: anything that touches `.circleci`
- refactor: internal refactor
- style: style change
- test: new tests

See https://www.conventionalcommits.org/en/v1.0.0/ for background.

The first line should also:

- be at most 89 characters long
- contain a description that is at most 72 characters long
- not end with sentence-ending punctuation
- start (after the type) with a lowercase imperative ("add", "fix")

-->

dzhu reviewed (approved):
I would say the cache buster should stay in the raw key for visibility and to match the other cache jobs. Also maybe additionally print out `$UID` and `$HOME` to the console for the sake of debugging. Otherwise, looks good, thanks!

dannysauer commented on .circleci/config.yml:
```diff
@@ -276,15 +276,21 @@ commands:
+          command: |
+            echo '<<pipeline.parameters.cache-buster>>' > /tmp/react-get-deps.cachekey
+            cat webui/react/package-lock.json >> /tmp/react-get-deps.cachekey
+            echo $UID $HOME >> /tmp/react-get-deps.cachekey
```
Are there $UIDs which share a $HOME? Feels like one or the other should be sufficiently unique. :)

dzhu replied:
Oh right, this should be the current directory. I started off thinking of using home directory, then switched to current directory, but forgot to update some parts of my brain.

EmilyBonar commented:
Anything still holding this up?

loksonarius commented:
Sorry, this fell off my plate while I was doing some clerical work.
Addressed comments in a new commit.
Enabling auto-merge -- this should go in ASAP.
